### PR TITLE
fix(component): file path traversal vulnerability.

### PR DESCRIPTION
**Pull request by rudra-superrr**
Description: This PR addresses a security vulnerability identified during penetration testing.

Problem: The SW360 web application allows users to download attachment bundles in ZIP format of attachments
stored in components. It was identified that this functionality is prone to a file path traversal vulnerability
when the application server is generating ZIP files with attachments that have ..\ in the filename.
The application server seems to copy the uploaded files to an application directory, which through
the path traversal vulnerability could allow attackers to write files to arbitrary folders in the server’s
filesystem.

Solution: Sanitize filenames at file system operations (file creation, ZIP generation) and not at database storage, preserving original user data while preventing path traversal attacks at the actual vulnerability points where filenames are used to create file paths or ZIP entries.

Testing:
Create a Test Component
Upload Test Files as attachment with Malicious fileNames(ex. ../test.txt or ../../tmp/test.txt)
Now download the attachment bundle from the component
Expected results: ../test.txt will be replaced by .._test.txt

### CODE REVIEW

File: libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java [hunk 1]
```diff
@@ -241,6 +241,41 @@ public static boolean isNotEmpty(Collection collection) {
...
+        }
+        
+        String sanitized = filename;
+        
+        // Remove forward slashes and backslashes (path separators)
+        sanitized = sanitized.replace("/", "_");
+        sanitized = sanitized.replace("\\", "_");
+        
+        // Remove parent directory references
+        sanitized = sanitized.replace("..", "_");
```
Comment (GMishx) [inline libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java]: Let's not do this as there can be some reasons to have `..` and on their own, they do no harm.

File: libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java [hunk 2]
```diff
@@ -241,6 +241,41 @@ public static boolean isNotEmpty(Collection collection) {
...
+        
+        // Remove leading dots to prevent hidden files
+        while (sanitized.startsWith(".")) {
+            sanitized = "_" + sanitized.substring(1);
+        }
+        
+        // If filename becomes empty after sanitization, use default
+        if (sanitized.trim().isEmpty()) {
+            return "attachment";
```
Comment (GMishx) [inline libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java]: If it is a default/constant, let's make it a static constant in the class and use in this return and the first return...

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java:1]: Please sanitize the filename in `public Attachment uploadAttachment(MultipartFile file, Attachment newAttachment, User sw360User)` function as well.

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by GMishx** [comment]
Can you please add a test case to the attachment controller test for this specific case?

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java
Comment (rudra-superrr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java:1]: done

File: libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java (hunk 2 above)
Comment (rudra-superrr) [reply libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java]: done

File: libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java (hunk 1 above)
Comment (rudra-superrr) [reply libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java]: done

**Comment by GMishx** [comment]
@bibhuti230185 please help test the PR

**Comment by bibhuti230185** [comment]
@GMishx  and @rudra-superrr . One very minor edge case with encoded URL separator fails but we can take the call if this is not a most relevant scenario.
## Verification Summary
- **Original issue**: Filenames like `../test.txt`, `../../test.txt`, `../../../tmp/test4.txt` allowed traversal during ZIP creation and file handling.
- **Applied fix**: Filenames are sanitized at file/ZIP creation points by replacing `/` and `\` with `_`.
- **Expected result**: Traversal strings become safe filenames (e.g., `../../tmp/test.txt` → `.._.._tmp_test.txt`), preventing writes outside the attachment/temp directories.
- **Known limitation (URL-encoded separators)**: `%2F` / `%5C` are not altered unless decoded before sanitization (e.g., `..%2F..%2Fetc%2Fpasswd`).
- **Minor**: `renameFile(...)` uses `lastIndexOf("/")`; Windows `\` paths can yield `-1` (prefer `getParent()` or `File.separator`).
- **Info**: redundant sanitization in `downloadAttachmentBundleWithContext` (safe, optional cleanup).
- **Manual checks performed**: traversal uploads (`../`, `../../`, `..\\`, mixed), absolute paths (`/etc/passwd`, `C:\\Windows\\System32\\config`) and bundle download verified sanitized ZIP entries.

### Source files checked (fix points)
- `libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/common/CommonUtils.java` (sanitizeFilename, lines 245-267)
- `libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/couchdb/AttachmentStreamConnector.java` (ZIP entry creation, lines 118-137)
- `rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java` (ZIP entry creation, lines 212-223; upload sanitize, lines 231-241; renameFile, lines 421-430)

**Review by GMishx — APPROVED** [review]
